DataDog/libdatadog
# [PROF-6762] Migrate Ruby gem release step to use new api key

**What does this PR do?**:

Update the Ruby release step to use the new api key.

**Motivation**:

Keep up with rubygems.org security best practices.

**Additional Notes**:

(N/A)

**How to test the change?**:

Because rubygems.org doesn't allow re-pushing the same version of a gem, you can actually run `docker-compose run push_to_rubygems` and observe that rubygems.org accepts the new key and command, but then complains that the current version of libdatadog (1.0.1) has already been pushed before.
